src/game/GameFSM.hpp

```diff
@@ -161,6 +165,26 @@ class GameFSM : public afsm::def::state_machine<GameFSM> {
 
                     // janitor is only active after the round limit was reached
                     if (state.getCurrentRound() >= matchConfig.getRoundLimit()) {
+                        if (!state.getJanitorCoordinates().has_value()) {
+                            auto randomField = GameLogicUtils::getRandomMapPoint(state, [&state](const auto &p) {
+                                // Random free/seat without character
+                                auto field = state.getMap().getField(p);
+                                if (field.getFieldState() != FieldStateEnum::BAR_SEAT
+                                    and field.getFieldState() != FieldStateEnum::FREE) {
+                                    // Wrong field type
+                                    return false;
+                                }
+                                return !GameLogicUtils::isPersonOnField(state, p);
+                            });
+
+                            if (!randomField.has_value()) {
+                                spdlog::critical("No field to place the janitor");
+                                std::exit(1);
```

Comment:
vielleicht lieber `throw`, dann könnte man vielleicht in der main oder so ein `while(true) { try...` oder so machen

Reply:
Mach ich, allerdings wäre es vermutlich am sinnvollsten direkt beim Einlesen des Szenarios zu prüfen, ob genug freie Felder vorliegen. Dann sollte der Fall an dieser Stelle nie eintreten.
Bleibt die Frage ob man bei zu wenig Feldern die Anzahl der NPCs reduziert oder das Szenario einfach ganz ablehnt.